airflow-core/src/airflow/api_fastapi/core_api/routes/public/task_instances.py

```diff
@@ -738,7 +761,13 @@ def _patch_ti_validate_request(
     map_index: int | None = -1,
     update_mask: list[str] | None = Query(None),
 ) -> tuple[DAG, list[TI], dict]:
-    dag = dag_bag.get_dag(dag_id)
+    try:
+        dag = dag_bag.get_dag(dag_id)
+    except DeserializationError as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"An error occurred while trying to deserialize DAG '{dag_id}': {e}",
+        )
```

Comment:
This shouldn't be copy pasted everywhere, I think factorizing that to the exception handler should be better. You can take a look at `init_error_handlers`

Reply:
IF you do that you'll also only have to test the exception handler, and not all the routes.